modules/v2/main.tf

```diff
@@ -0,0 +1,285 @@
+            value = env.value
+          }
+        }
+
+        dynamic "env" {
+          for_each = containers.value.env_secret_vars
+          content {
+            name = env.key
+            value_source {
+              secret_key_ref {
+                secret  = env.value.secret
+                version = env.value.version
+              }
+            }
+          }
+        }
+
+        dynamic "volume_mounts" {
+          for_each = containers.value.volume_mounts
+          content {
+            name       = volume_mounts.value["name"]
+            mount_path = volume_mounts.value["mount_path"]
+          }
+        }
+      }
+    } // containers
+
+    dynamic "volumes" {
+      for_each = var.volumes
+      content {
+        name = volumes.value["name"]
+
+        dynamic "secret" {
+          for_each = volumes.value.secret[*]
+          content {
+            secret = secret.value["secret"]
+            items {
+              path    = secret.value.items["path"]
+              version = secret.value.items["version"]
+              mode    = secret.value.items["mode"]
+            }
+          }
+        }
+
+        dynamic "cloud_sql_instance" {
+          for_each = volumes.value.cloud_sql_instance[*]
+          content {
+            instances = cloud_sql_instance.value["instances"]
+          }
+        }
+        dynamic "empty_dir" {
+          for_each = volumes.value.empty_dir[*]
+          content {
+            medium     = empty_dir.value["medium"]
+            size_limit = empty_dir.value["size_limit"]
+          }
+        }
+        dynamic "gcs" {
+          for_each = volumes.value.gcs[*]
+          content {
+            bucket    = gcs.value["medium"]
+            read_only = gcs.value["size_limit"]
+          }
+        }
+        dynamic "nfs" {
+          for_each = volumes.value.nfs[*]
+          content {
+            server    = nfs.value["medium"]
+            path      = nfs.value["path"]
+            read_only = nfs.value["size_limit"]
+          }
+        }
+      }
+    }
+  } // template
+
+  annotations      = var.service_annotations
+  client           = var.client.name
+  client_version   = var.client.version
+  ingress          = var.ingress
+  launch_stage     = var.launch_stage
+  custom_audiences = var.custom_audiences
+
+  dynamic "binary_authorization" {
+    for_each = var.binary_authorization[*]
+    content {
+      breakglass_justification = binary_authorization.value.breakglass_justification
+      use_default              = binary_authorization.value.use_default
+    }
+  }
+
+  dynamic "scaling" {
+    for_each = var.service_scaling[*]
+    content {
+      min_instance_count = scaling.value.min_instance_count
+    }
+  }
+
+  dynamic "traffic" {
+    for_each = var.traffic
+    content {
+      percent  = traffic.value.percent
+      type     = traffic.value.type
+      revision = traffic.value.revision
+      tag      = traffic.value.tag
+    }
+  }
+}
```

Comment:
Do we need to ignore chnage any annotations added by the service like
https://github.com/GoogleCloudPlatform/terraform-google-cloud-run/blob/b59b700a7a5d3c15847036792db88df7eb9503db/main.tf#L192

Reply:
I guess these are not needed as they will be part of [effective_annotations](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloud_run_v2_service#effective_annotations) and we can add them later if really needed.